src/SFA.DAS.Admin.Aan.Web/Controllers/ManageEvent/GuestSpeakersController.cs

```diff
@@ -0,0 +1,164 @@
+        _sessionService = sessionService;
+        _addGuestSpeakerValidator = addGuestSpeakerValidator;
+        _hasGuestSpeakersValidator = hasGuestSpeakersValidator;
+    }
+    [HttpGet]
+    [Route("question", Name = RouteNames.ManageEvent.EventHasGuestSpeakers)]
+    public IActionResult GetHasGuestSpeakers()
+    {
+        var sessionModel = _sessionService.Get<EventSessionModel>();
+        var model = GetViewModelHasGuestSpeakers(sessionModel);
+        return View(HasGuestSpeakersViewPath, model);
+    }
+
+    [HttpPost]
+    [Route("question", Name = RouteNames.ManageEvent.EventHasGuestSpeakers)]
+    public IActionResult PostHasGuestSpeakers(HasGuestSpeakersViewModel submitModel)
+    {
+        var sessionModel = _sessionService.Get<EventSessionModel?>();
+        sessionModel!.HasGuestSpeakers = submitModel.HasGuestSpeakers;
+
+        var result = _hasGuestSpeakersValidator.Validate(submitModel);
+
+        if (!result.IsValid)
+        {
+
+            result.AddToModelState(ModelState);
+            return View(HasGuestSpeakersViewPath, GetViewModelHasGuestSpeakers(sessionModel));
+        }
+
+        _sessionService.Set(sessionModel);
+
+        if (sessionModel.HasGuestSpeakers == true) return RedirectToRoute(RouteNames.ManageEvent.GuestSpeakerList);
+        return RedirectToRoute(RouteNames.ManageEvent.EventHasGuestSpeakers);
+    }
+
+    [HttpGet]
+    [Route("add", Name = RouteNames.ManageEvent.GuestSpeakerAdd)]
+    public IActionResult GetAddGuestSpeaker(GuestSpeakerAddViewModel model)
+    {
+        var augmentedModel = GetGuestSpeakerAddViewModel(model);
+
+        return View(GuestSpeakerAddViewPath, augmentedModel);
+    }
+
+
+    [HttpGet]
+    [Route("delete", Name = RouteNames.ManageEvent.GuestSpeakerDelete)]
+    public IActionResult DeleteGuestSpeaker(int id)
+    {
+        var sessionModel = _sessionService.Get<EventSessionModel?>();
+        var currentGuestList = sessionModel!.GuestSpeakers;
+        if (currentGuestList.Any())
+        {
+            var removeItem = currentGuestList.First(x => x.Id == id);
+            currentGuestList.Remove(removeItem);
+        }
+
+        sessionModel.GuestSpeakers = currentGuestList;
+        _sessionService.Set(sessionModel);
+        return RedirectToRoute(RouteNames.ManageEvent.GuestSpeakerList);
+    }
+
+    [HttpPost]
+    [Route("add", Name = RouteNames.ManageEvent.GuestSpeakerAdd)]
+    public IActionResult PostAddGuestSpeaker(GuestSpeakerAddViewModel submitModel)
+    {
+        var result = _addGuestSpeakerValidator.Validate(submitModel);
+
+        if (!result.IsValid)
+        {
+
+            result.AddToModelState(ModelState);
+            return View(GuestSpeakerAddViewPath, GetGuestSpeakerAddViewModel(submitModel));
+        }
+
+        var sessionModel = _sessionService.Get<EventSessionModel?>();
+        var currentGuestList = sessionModel!.GuestSpeakers;
+
+        var id = currentGuestList.Any() ? currentGuestList.Max(x => x.Id) + 1 : 1;
```

Comment:
This is superfluous id which is not going to the backend, I suppose you are using this to manage guests in the memory. Wouldn't it be better if you use guid so you dont have to calculate id?

Reply:
I have used id to make the add/delete process easier.  In addition, when we start linking up to edit real details, we will be using GuestSpeaker as the basis, which has a PK of 'id' which is an int. Therefore, building in an int 'id' into the GuestSpeaker list has >1 benefit.